overlord/snapstate/handlers.go

```diff
@@ -1236,6 +1282,7 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) (err error) {
 		if unlinkErr != nil {
 			t.Errorf("cannot cleanup failed attempt at making snap %q available to the system: %v", snapsup.InstanceName(), unlinkErr)
 		}
+		notifyLinkParticipants(t, snapsup.InstanceName())
```

Comment:
we have a preexisting bug where it seems nowadays we can get here after we have Set the snapstate, this means we can end up calling the link participant with Active = true, but we just unlinked the snap again...

bit of a mess that we need to untangle somehow

Reply:
I understand this is solved by your branch which is now merged here.